tests/integration/test_filters.py

```diff
@@ -282,3 +282,37 @@ def test_ignore_transactions_filters_are_applied(
     else:
         event, _ = transactions_consumer.get_event()
         assert event["transaction"] == transaction_name
+
+
+def test_global_filters_drop_events(
```

Comment:
An additional test that has a more complex scenario of combining global and project state would be nice.

Can be one test that tests a bunch of things like:
- Disabled in the global config but enabled in project state
- Enabled in global config, disabled in project state
- Enabled in both
- Disabled in both

Could also include version mismatches and differences in the tests but you already have separate tests for that 👍

Reply:
I'll cover these cases with unit tests, as they are validating the iterator's implementation logic.

Reply:
Fair enough, although that should also be nicely doable with pytest and parameterized tests.

Reply:
It's possible, but the unit tests are more scoped, easier to spot, faster to run, and less surface area to be flaky 😄